src/sysext/sysext.c

```diff
@@ -1029,6 +1041,13 @@ static int parse_argv(int argc, char *argv[]) {
                                 return r;
                         break;
 
+                case ARG_NOEXEC:
+                        r = parse_boolean_argument("--noexec", optarg, (bool *)&arg_noexec);
```

Comment:
Oops, arg_noexec is tristate. Please do not pass the pointer. IIRC, (not sure, I am from phone) there is parse_tristate_argument() or so.

Reply:
Then, simply assign r for arg_noexec on success.